packages/studio/src/components/ErrorComponentPreview.tsx

```diff
@@ -1,32 +1,63 @@
-import { ErrorComponentState } from "@yext/studio-plugin";
-import { useCallback, useMemo, useRef, useState } from "react";
+import {
+  useCallback,
+  useMemo,
+  useRef,
+  useState,
+  useEffect,
+  Dispatch,
+  SetStateAction,
+} from "react";
 import { v4 } from "uuid";
 import ErrorBoundary from "./common/ErrorBoundary";
-import { Tooltip } from "react-tooltip";
+import { ErrorComponentState } from "@yext/studio-plugin";
+import { ITooltip } from "react-tooltip";
 
 export default function ErrorComponentPreview(props: {
   errorComponentState: ErrorComponentState;
   element: JSX.Element | null;
+  setTooltipProps: Dispatch<SetStateAction<ITooltip>>;
 }) {
-  const { errorComponentState, element } = props;
+  const { errorComponentState, element, setTooltipProps } = props;
   const containerRef = useRef<HTMLDivElement>(null);
   const uniqueId = useMemo(() => v4(), []);
   // We cannot use the uuid on ErrorComponentState due to
   // repeaters re-using the same uuid
   const anchorId = `ErrorComponentState-${uniqueId}`;
   const [tooltipOpen, setTooltipOpen] = useState(false);
+  const closeTooltip = () => setTooltipOpen(false);
   const onMouseEnter = useCallback(() => {
     setTooltipOpen(true);
   }, []);
-  const onMouseLeave = useCallback(() => {
-    setTooltipOpen(false);
+  const onMouseLeave = useCallback(closeTooltip, []);
+
+  useEffect(() => {
+    document
+      .querySelector("iframe")
+      ?.contentDocument?.addEventListener("scroll", closeTooltip);
+    document.addEventListener("scroll", closeTooltip);
```

Comment:
when would this listener's callback have an effect? in order for the scroll to occur on the top-level document rather than the iframe, the mouse would have to be outside the iframe, in which case the tooltip would already be closed because of the onMouseLeave, right?

Reply:
Oh so u can oddly scroll in the outer window even while ur in the preview panel.

Reply:
ah like if you're already at the bottom of the preview panel and continue scrolling down, or if the preview panel is not scrollable but the outer window is. got it!